src/types/terminal.d.ts

```diff
@@ -0,0 +1,20 @@
+import { ReactElement } from 'react';
+
+export interface TerminalFile {
+  id: string;
+  title: string;
+  type: string;
+  content: JSX.Element;
+}
+
+export interface TerminalDirectory {
+  id: string;
+  title: string;
+  type: string;
+  children: (TerminalFile | TerminalDirectory)[];
+}
```

Comment:
> Terminal의 contents가 file, directory에 따라 타입이 나뉘어져 있어서, 특정 로직에서 as 키워드를 이용해 타입 단언을 이용하고 있습니다.(ex. terminal/useTerminal/displayContent, utils/terminal.ts/getCurrentChildren)
> 
> 이처럼 배열에 여러 타입의 요소가 있을 때, 타입 단언을 사용하지 않고 문제를 해결할 수 있는 좋은 방법이 있을까요? 🤔

아래 처럼 TerminalFile, TerminalDirectory 의 type 을 리터럴 타입으로 정의하면 타입스크립트가 배열에 있는 요소를 추론할 수 있을 것 같아요. 😃 

```suggestion
export interface TerminalFile {
  id: string;
  title: string;
  type: 'file';
  content: JSX.Element;
}

export interface TerminalDirectory {
  id: string;
  title: string;
  type: 'directory';
  children: (TerminalFile | TerminalDirectory)[];
}
```

Reply:
@minjongbaek 
리터럴 타입으로 범위를 좁혀줬더니, if문으로도 분기가 가능해지네요...! 하나 배웠습니다 감사합니다!😆https://github.com/shaw-kee/shawkee-os/pull/28/commits/0b0aa5583bdab9da6745884458a36cc07df850b7